cmd/guard-service/main.go

```diff
@@ -45,12 +49,74 @@ type learner struct {
 	pileLearnTicker *utils.Ticker
 }
 
+// Common method used for parsing ns, sid, cmFlag from all requests
+func (l *learner) baseHandler(query url.Values) (ns string, sid string, cmFlag bool, record *serviceRecord) {
+	cmFlagSlice := query["cm"]
+	sidSlice := query["sid"]
+	nsSlice := query["ns"]
+	if len(sidSlice) != 1 || len(nsSlice) != 1 || len(cmFlagSlice) > 1 {
+		log.Infof("baseHandler wrong data sid %d ns %d cmFlag %d", len(sidSlice), len(nsSlice), len(cmFlagSlice))
+		return
+	}
+	sid = utils.Sanitize(sidSlice[0])
+	ns = utils.Sanitize(nsSlice[0])
+	if len(cmFlagSlice) > 0 {
+		cmFlag = (cmFlagSlice[0] == "true")
+	}
+	if strings.HasPrefix(sid, "ns-") {
+		log.Infof("baseHandler illegal sid")
+		sid = ""
+		return
+	}
+
+	record = l.services.get(ns, sid, cmFlag)
+	return
+}
+
 func (l *learner) fetchConfig(w http.ResponseWriter, req *http.Request) {
```

Comment:
Same as previous function. return error and deal with it in the caller code. This allows more flexibility for retries etc.